src/GetStmtHierarchy.cpp

```diff
@@ -0,0 +1,724 @@
+
+        int curr_node = curr_node_ID;
+        for (auto &e : args) {
+            curr_node_ID = curr_node;
+            e.accept(this);
+        }
+        html << close_node();
+    }
+
+    else {
+        std::vector<Expr> args = op->vectors;
+        for (int i : op->indices) {
+            args.emplace_back(i);
+        }
+        html << open_node(op, "Shuffle");
+
+        int curr_node = curr_node_ID;
+        for (auto &e : args) {
+            curr_node_ID = curr_node;
+            e.accept(this);
+        }
+        html << close_node();
+    }
+}
+void GetStmtHierarchy::visit(const VectorReduce *op) {
+    html << open_node(op, "vector_reduce");
+
+    int curr_node = curr_node_ID;
+    stringstream op_op;
+    op_op << op->op;
+    html << node_without_children(nullptr, op_op.str());
+
+    curr_node_ID = curr_node;
+    op->value.accept(this);
+
+    html << close_node();
+}
+void GetStmtHierarchy::visit(const Prefetch *op) {
+    internal_error << "\n"
+                   << "GetStmtHierarchy: Prefetch is not supported yet. \n\n";
+}
+void GetStmtHierarchy::visit(const Fork *op) {
+    internal_error << "\n"
+                   << "GetStmtHierarchy: Fork is not supported yet. \n\n";
+}
+void GetStmtHierarchy::visit(const Acquire *op) {
+    html << open_node(op, "acquire");
+
+    int curr_node = curr_node_ID;
+    op->semaphore.accept(this);
+
+    curr_node_ID = curr_node;
+    op->count.accept(this);
+
+    html << close_node();
+}
+void GetStmtHierarchy::visit(const Atomic *op) {
+    if (op->mutex_name.empty()) {
+        html << node_without_children(op, "atomic");
+    } else {
+        html << open_node(op, "atomic");
+        html << node_without_children(nullptr, op->mutex_name);
+        html << close_node();
+    }
+}
+
+string GetStmtHierarchy::generate_stmt_hierarchy_js() {
+    stringstream stmt_hierarchy_js;
+
+    stmt_hierarchy_js
+        << "\n// stmtHierarchy JS\n"
+        << "for (let i = 1; i <= " << stmt_hierarchy_tooltip_count << "; i++) { \n"
+        << "    const button = document.getElementById('stmtHierarchyButtonTooltip' + i); \n"
+        << "    const tooltip = document.getElementById('stmtHierarchyTooltip' + i); \n"
+        << "    button.addEventListener('mouseenter', () => { \n"
+        << "        showTooltip(button, tooltip); \n"
+        << "    }); \n"
+        << "    button.addEventListener('mouseleave', () => { \n"
+        << "        hideTooltip(tooltip); \n"
+        << "    } \n"
+        << "    ); \n"
+        << "    tooltip.addEventListener('focus', () => { \n"
+        << "        showTooltip(button, tooltip); \n"
+        << "    } \n"
+        << "    ); \n"
+        << "    tooltip.addEventListener('blur', () => { \n"
+        << "        hideTooltip(tooltip); \n"
+        << "    } \n"
+        << "    ); \n"
+        << "} \n";
+
+    return stmt_hierarchy_js.str();
+}
+
+const string GetStmtHierarchy::stmt_hierarchy_css = "\n \
```

Comment:
(1) you definitely want a raw string literal here
(2) declaring this as `const char *` or `const char[]` instead of `std::string` would avoid unnecessary memory allocations